gnosis/gp-swap-ui
# [FIX] Buy order incorrect numbers + use slippage adjusted amounts

Closes #369 and closes #370 

Logic wasn't sending the SELL or BUY amounts with slippage applied to the API. It was sending the expected amounts with no adjustment for slippage - this PR sends the correct values.

To test: (RINKEBY::WETH for USDC)
`SELL`
1. Set a high slippage amount, say 49% and confirm it shows that on swap
2. Set WETH sell to `0.0005`
3. Make sure all params are correct: minimum received should be the displayed `To` amount computed with slippage:
assuming `To` amount shows `0.251709`, then:
```
0.251709 * (1 + 0.49) = 0.1689 USDC
```
4. Submit the order
5. Check that the confirmation modal screens show correct numbers
6. Click `Confirm swap`
7. looks at the MM popup and make sure that the:
  a. `sellAmount` + `feeAmount` = the amount you initially input (it will be shown in atoms)
  b. that the buyAmount is the SLIPPAGE adjusted amount we calculated in step 3

`BUY`
1. Set a high slippage amount, say 49% and confirm it shows that on swap
2. Set USDC buy to `0.251709`
3. Make sure all params are correct: maximum sold should be the displayed `From` amount computed with slippage:
assuming `From` amount shows `0.000504998`, then:
```
0.000504998 * (1 + 0.49) = 0.0007524 WETH
```
4. Submit the order
5. Check that the confirmation modal screens show correct numbers
6. Click `Confirm swap`
7. looks at the MM popup and make sure that the:
  a. `sellAmount` + `feeAmount` = the slippage adjusted sell amount in atoms
  b. that the buyAmount is the exact amount you input initially in step 2

For both: check that the toast notification messages show the fee adjusted amounts and the slippage adjusted amounts correctly

EDIT: leandro schooled me in maths
